Kernel/include/kernelTypes.h

```diff
@@ -47,10 +50,17 @@ typedef struct {
 /**
  * @brief Represents a semaphore.
  */
-typedef int TSem;
+typedef int8_t TSem;
 
 typedef struct {
-    const char* name;
+    int value;
+    int linkedProcesses; 
+    char name[MAX_NAME_LENGTH+1];
+    //Diference with pipes: allways save space for -1, so that the only condition
+    // that needs to be checked is thatone.
+    // Otherwise, the user shoud check if the pid is -1 or if they have reached MAX_PID_ARRAY_LENGHT
+    // TODO: Check what to change so that to be consistent
+    TPid waitingProcesses[MAX_PID_ARRAY_LENGTH+1];
```

Comment:
We should keep consistency. @ichayer You get to break the tie and decide which method stays lol

Reply:
Its just a suggestion, all i did was based on your pipes implementation. If you think your implementation its better, i can align my code to that

Reply:
Let's choose the simplest and consistent solution

Reply:
I'll update the listPipes to match this.
```suggestion
    TPid waitingProcesses[MAX_PID_ARRAY_LENGTH+1];
```